assets/js/base/context/cart-checkout/billing/constants.js

```diff
@@ -8,7 +8,13 @@
  */
 import { getSetting } from '@woocommerce/settings';
 
-const checkoutData = getSetting( 'checkoutData', {} );
+const defaultCheckoutData = {
+	billing_address: {},
+};
+
+// Checkout data may be null in editor/admin.
+const checkoutData =
+	getSetting( 'checkoutData', defaultCheckoutData ) || defaultCheckoutData;
```

Comment:
Shouldn’t the first default be enough?

Reply:
Unfortunately it's coming back null. Not sure if this is a bug with `getSetting`, but that's what's happening. I agree would be better to simplify this.

I just tried `const checkoutData = getSetting( 'checkoutData', defaultCheckoutData );` and saw the js error - `Cannot read property 'billing_address' of null`.

Reply:
yeah it only seems it check if the value is set, not the value by its own
https://github.com/woocommerce/woocommerce-gutenberg-products-block//blob/a1717cbbe811466fefac04b141be60128e56c356/assets/js/settings/shared/get-setting.js#L21-L27
we can either pass a filter function but that just complicates it, or we can add a check in PHP when we're defining this setting

Reply:
something like this
https://github.com/woocommerce/woocommerce-gutenberg-products-block//blob/ee73458babfb6ef8d35987491d2f5900280f6255/src/BlockTypes/Checkout.php#L74-L76